lib/elixir/lib/kernel.ex

```diff
@@ -4117,6 +4117,64 @@ defmodule Kernel do
   end
 
   @doc """
+  Generates a macro suitable for use in guard expressions.
+
+  It raises at compile time if the definition uses expressions that aren't
+  allowed in guards, and otherwise creates a macro that can be used both inside
+  or outside guards.
+
+  ## Example
+
+      defmodule Integer.Guards do
+        defguard is_even(value) when is_integer(value) and rem(value, 2) == 0
+      end
+
+      defmodule Integer.Utils do
+        import Integer.Guards
+
+        def is_even_and_large?(value) when is_even(value) and value > 100, do: true
+        def is_even_and_large?(_), do: false
+
+        def is_even_and_small?(value) do
+          if is_even(value) and value <= 100, do: true, else: false
+        end
```

Comment:
The examples I provided are awkward, but try to concisely showcase that `is_even/1` can be used both inside and outside guards. I don't think there's an opportunity to do both when implementing collatz, though. Definitely needs a better example though.